patch: make groups scope optional to support azure with OIDC (#9773) (#9778)
determined-ai/determined: docs/reference/deploy/master-config-reference.rst

For Kubernetes installations, if you define additional resource managers, the resource manager
specified under the primary resource_manager key here is considered the default.

``cluster_name``
================

Optional for single resource manager configurations. Required for multiple resource manager
(Multi-RM) configurations. Specifies the resource manager's associated cluster name. This references
the cluster on which a Determined deployment is running. Defaults to ``default`` if not specified.
For Kubernetes installations with additional resource managers, ensure unique names for all resource
managers in the cluster.

> **NOTE:** ``resource_manager.cluster_name`` is separate from the ``cluster_name`` field of the

Reviewer: ```suggestion
.. note::

The `resource_manager.cluster_name` is distinct from the `cluster_name` field in the master configuration, which provides a readable name for the Determined deployment.
```

The name of the attribute passed in through the claim from the OIDC provider used to set the user's
display name in Determined.

``always_redirect``
===================

> Specifies if this OIDC provider should be used for authentication, bypassing the standard Determined

Reviewer: ```suggestion
Specifies whether this OIDC provider should be used for authentication, bypassing the standard Determined
```

> sign-in page. This redirection persists unless the user explicitly signs out within the WebUI. If an

Reviewer: ```suggestion
sign-in page. This redirection persists unless the user explicitly signs out via the WebUI. If an
```

> SSO user attempts to use an expired session token, they are directly redirected to the SSO provider

Reviewer: ```suggestion
SSO user attempts to use an expired session token, they will be redirected to the SSO provider
```

and returned to the requested page after authentication.

``exclude_groups_scope``
========================

> Specifies if the groups scope should be excluded for this OIDC provider. For most OIDC providers

Reviewer: ```suggestion
Indicates whether the groups scope should be excluded for this OIDC provider. For most OIDC providers
```

> such as Okta, this should be false (or blank) if you'd like to provision group memberships. But for

Reviewer: ```suggestion
such as Okta, this should be set to false (or blank) if you want to provision group memberships. However, for
```

The claim name from the SAML provider used to set the user's display name in Determined.

``always_redirect``
===================

> Specifies if this SAML provider should be used for authentication, bypassing the standard Determined

Reviewer: ```suggestion
Specifies whether this SAML provider should be used for authentication, bypassing the standard Determined

> sign-in page. This redirection persists unless the user explicitly signs out within the WebUI. If a

Reviewer: ```suggestion
sign-in page. This redirection persists unless the user explicitly signs out via the WebUI. If a
```

> SSO user attempts to use an expired session token, they are directly redirected to the SAML provider

Reviewer: ```suggestion
SSO user attempts to use an expired session token, they will be redirected to the SAML provider
```
